web-components/src/components/cards/PostCard/PostCard.tsx

```diff
@@ -0,0 +1,331 @@
+      {name}
+    </Body>
+  );
+};
+
+const NameWithRoleAndTimestamp = ({
+  name,
+  authorRole,
+  timestamp,
+}: {
+  name: string;
+  authorRole?: string;
+  timestamp: string;
+}): JSX.Element => (
+  <Box
+    sx={{
+      ml: 3,
+    }}
+  >
+    <Box>
+      <Subtitle size="sm" sx={{ display: 'inline' }}>
+        {name}
+      </Subtitle>
+      <Box
+        borderRadius={1}
+        sx={{
+          px: 1.5,
+          py: 0.5,
+          ml: 2,
+          backgroundColor: theme => theme.palette.grey['300'],
+          display: 'inline-block',
+          fontSize: '10px',
+          color: theme => theme.palette.grey['500'],
+          textTransform: 'uppercase',
+          fontWeight: 800,
+          letterSpacing: '1px',
+        }}
+      >
+        {authorRole}
+      </Box>
+    </Box>
+    <Box
+      sx={theme => ({
+        color: theme.palette.grey['700'],
+        fontSize: '12px',
+        mt: 1.5,
+      })}
+    >
+      {formatDate(timestamp)}
+    </Box>
+  </Box>
+);
+
+export default function PostCard({
+  title,
+  description,
+  imgSrc,
+  geojson,
+  isGISFile,
+  mapboxToken,
+  imageStorageBaseUrl,
+  apiServerUrl,
+  author,
+  authorRole,
+  signer,
+  timestamp,
+  isPrivate,
+  isAdmin,
+  numberOfFiles,
+  handleClickFile,
+  adminMenuItems,
+  handleClickShare,
+}: PostCardProps): JSX.Element {
+  const { classes } = useStyles();
+
+  const authorWithNameRaw: User = {
+    ...author,
+    nameRaw: (
+      <NameWithRoleAndTimestamp
+        name={author.name}
+        authorRole={authorRole}
+        timestamp={timestamp}
+      />
+    ),
+  };
+
+  const signerWithNameRaw: User | undefined = signer && {
+    ...signer,
+    nameRaw: <SignerName name={signer.name} />,
+  };
+
+  return (
+    <Card className={classes.root} sx={{ p: [4, 8] }} borderRadius="10px">
+      <ActionButton
+        isAdmin={isAdmin}
+        adminMenuItems={adminMenuItems}
+        onClick={handleClickShare}
+      />
+      <Grid
+        container
+        sx={{ flexWrap: ['wrap-reverse', 'nowrap'], position: 'relative' }}
+      >
+        <Grid xs={12} sm={7} item sx={{ pb: [4.5, 0], pr: [0, 2] }}>
+          <Subtitle size="xl" mb={2.75}>
+            {title}
+          </Subtitle>
+          <UserInfo
+            size="md"
+            user={authorWithNameRaw}
+            sx={{ display: 'flex', alignItems: 'center' }}
+            nameHasPadding={false}
+          />
+          <Box sx={{ paddingInlineEnd: 2, paddingBlockStart: 4 }}>
+            <Body
+              size="lg"
+              mobileSize="sm"
+              sx={{ pb: 1.5 }}
+              className={classes.description}
+            >
+              {description}
+            </Body>
+          </Box>
+          {signerWithNameRaw && (
+            <Box sx={{ display: 'flex', alignItems: 'center', mt: 3 }}>
+              <VerifiedIcon color="white" fontSize="medium" hasFill />
+              <Body
+                size="sm"
```

Comment:
this should be xs on both desktop and mobile:
```suggestion
                size="xs" mobileSize="xs"
```